### PR TITLE
Fix joystick functionality for ChibiOS and OTG (Blackpill)

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

This fixes an issue I was seeing where no joystick reports were being sent on my Blackpill testbed, but worked fine on other ARM boards (eg. Proton-C and Bluepill). Turns out the joystick endpoint setup (and likely digitizer too, but I cannot test it) was incorrect.

A bunch of empty IN notification callbacks were also removed.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: tmk_core/protocol/chibios/usb_main.c [hunk 1]
```diff
@@ -173,6 +173,42 @@ static const USBEndpointConfig shared_ep_config = {
...
+    NULL,                   /* OUT notification callback */
+    JOYSTICK_EPSIZE,        /* IN maximum packet size */
+    0,                      /* OUT maximum packet size */
+    &joystick_ep_state,     /* IN Endpoint state */
+    NULL,                   /* OUT endpoint state */
+    usb_lld_endpoint_fields /* USB driver specific endpoint fields */
+};
+#endif
+
+#ifdef DIGITIZER_ENABLE
```
Comment (sigprof) [inline tmk_core/protocol/chibios/usb_main.c:194]:
```suggestion
#if defined(DIGITIZER_ENABLE) && !defined(DIGITIZER_SHARED_EP)
```

File: tmk_core/protocol/chibios/usb_main.c [hunk 2]
```diff
@@ -482,6 +496,12 @@ static void usb_event_cb(USBDriver *usbp, usbevent_t event) {
 #endif
 #ifdef SHARED_EP_ENABLE
             usbInitEndpointI(usbp, SHARED_IN_EPNUM, &shared_ep_config);
+#endif
+#ifdef JOYSTICK_ENABLE
+            usbInitEndpointI(usbp, JOYSTICK_IN_EPNUM, &joystick_ep_config);
+#endif
+#ifdef DIGITIZER_ENABLE
```
Comment (sigprof) [inline tmk_core/protocol/chibios/usb_main.c:503]:
```suggestion
#if defined(DIGITIZER_ENABLE) && !defined(DIGITIZER_SHARED_EP)
```

**Review by sigprof — CHANGES_REQUESTED** [review]
Looks this needs some fixes to support `DIGITIZER_SHARED_EP` properly (not actually tested).

**Comment by sigprof** [comment]
There is a `digitizer` keymap for onekey that should send something, but I never actually tried it.

**Review by drashna — APPROVED** [review]
Doesn't seem to break anything, at least.

**Comment by fauxpark** [comment]
Looks like Blackpill onekey digitizer keymap already works on develop (somehow), but this PR doesn't break it 👍

**Pull request merged**